### PR TITLE
stop bubbling modexp call success, assert instead

### PR DESCRIPTION
**Pull request by nalinbhardwaj**
not in audit, but good fix to clean up.

### CODE REVIEW

File: script/deploy.sh [hunk 1]
```diff
@@ -1,16 +1,16 @@
 
 # Deploy contract
-forge script DeployScript --via-ir --optimizer-runs=999999 --rpc-url $RPC_URL --broadcast --private-key $PRIVATE_KEY
+forge script DeployScript --rpc-url $RPC_URL --broadcast --private-key $PRIVATE_KEY --verify --etherscan-api-key $ETHERSCAN_API_KEY
```
Comment (dcposch) [inline script/deploy.sh:3]: is this because we're setting optimizer runs in foundry.toml?

File: src/P256Verifier.sol [hunk 2]
```diff
@@ -390,15 +385,15 @@ contract P256Verifier {
      * by Fermat's little theorem, compute u^(f-2) mod f using modexp precompile
      * Assume f != 0.
      */
-    function modInv(uint256 u, uint256 f, uint256 minus_2modf) internal view returns (uint256 result, bool success) {
+    function modInv(uint256 u, uint256 f, uint256 minus_2modf) internal view returns (uint256 result) {
```
Comment (dcposch) [inline src/P256Verifier.sol:401]: nice cleanup

File: script/deploy.sh (hunk 1 above)
Comment (nalinbhardwaj) [reply script/deploy.sh:3]: yep, wanted to keep variables like that to one place.

**Review by dcposch — APPROVED** [review]
LGTM.

Recommend:
- reverting `README/broadcast` since those will change again
- shipping https://github.com/daimo-eth/p256-verifier/pull/16 first
- then we can make a PR into `audit-fix-1` upgrading the submodule
- ...we can talk about this PR during our next check in with Veridise to maybe include in the followup review.

**Pull request merged**